test/xref_runner_SUITE.erl

```diff
@@ -380,3 +381,97 @@ check_with_config_file(_Config) ->
+    end,
+
+  WriteTestConfig =
+    fun(Config) ->
+      file:write_file("test-xref.config", io_lib:format("~p.", [Config]))
+    end,
+
+  OldCwd = file:get_cwd(),
+
+  try
+    ct:comment("Empty config works as if there is no config"),
+    ok = WriteConfig([]),
+    [] = xref_runner:main([]),
+
+    ct:comment("Empty list of options works as if there is no config"),
+    ok = WriteConfig([{xref, []}]),
+    [] = xref_runner:main([]),
+
+    ct:comment("With the proper dir, but no checks, runs all checks"),
+    Path = filename:dirname(code:which(ignore_xref)),
+    ok = WriteConfig([{xref, [{config, #{dirs => [Path]}}]}]),
+    AllResults = xref_runner:main([]),
+    [_|_] = [1 || #{check := undefined_function_calls} <- AllResults],
+    [_|_] = [1 || #{check := undefined_functions} <- AllResults],
+    [_|_] = [1 || #{check := locals_not_used} <- AllResults],
+    [_|_] = [1 || #{check := exports_not_used} <- AllResults],
+    [_|_] = [1 || #{check := deprecated_function_calls} <- AllResults],
+    [_|_] = [1 || #{check := deprecated_functions} <- AllResults],
+
+    ct:comment("With the proper dir, with checks, runs only those checks"),
+    Path = filename:dirname(code:which(ignore_xref)),
+    ok =
+      WriteConfig(
+        [ { xref
+          , [ {config, #{dirs => [Path]}}
+            , {checks, [locals_not_used, exports_not_used]}
+            ]
+          }
+        ] ),
+    SomeResults = xref_runner:main([]),
+    [] = [1 || #{check := undefined_function_calls} <- SomeResults],
+    [] = [1 || #{check := undefined_functions} <- SomeResults],
+    [_|_] = [1 || #{check := locals_not_used} <- SomeResults],
+    [_|_] = [1 || #{check := exports_not_used} <- SomeResults],
+    [] = [1 || #{check := deprecated_function_calls} <- SomeResults],
+    [] = [1 || #{check := deprecated_functions} <- SomeResults],
+
+    ct:comment("With the proper dir, with checks == [], runs no check"),
+    Path = filename:dirname(code:which(ignore_xref)),
+    ok =
+      WriteConfig(
+        [ { xref
+          , [ {config, #{dirs => [Path]}}
+            , {checks, []}
+            ]
+          }
+        ] ),
+    [] = xref_runner:main([]),
+
+    ct:comment("With the proper dir, with checks, specifying xref.config path"),
+    Path = filename:dirname(code:which(ignore_xref)),
+    ok =
+      WriteTestConfig(
+        [ { xref
+          , [ {config, #{dirs => [Path]}}
+            , {checks, [locals_not_used, exports_not_used]}
+            ]
+          }
+        ] ),
+    SomeResults1 = xref_runner:main("-c test-xref.config"),
+    [] = [1 || #{check := undefined_function_calls} <- SomeResults],
+    [] = [1 || #{check := undefined_functions} <- SomeResults],
+    [_|_] = [1 || #{check := locals_not_used} <- SomeResults],
+    [_|_] = [1 || #{check := exports_not_used} <- SomeResults],
+    [] = [1 || #{check := deprecated_function_calls} <- SomeResults],
+    [] = [1 || #{check := deprecated_functions} <- SomeResults],
+
+    {comment, ""}
```

Comment:
You're not testing `xref_runner:main("-h")` nor `xref_runner:main("--config some-unexistent.config")` neither `xref_runner:main("something that's clearly not a valid option")`.